docker/buoy/Dockerfile

```diff
@@ -0,0 +1,59 @@
+FROM ros:galactic-ros-base
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Useful tools
+RUN apt update \
+ && apt install -y \
+        apt-utils \
+        build-essential \
+        curl \
+        lsb-release \
+        wget \
+ && apt clean
+
+# Setup and install ignition fortress, colcon and rosdep
+RUN /bin/sh -c 'wget https://packages.osrfoundation.org/gazebo.gpg -O /usr/share/keyrings/pkgs-osrf-archive-keyring.gpg' \
+  && /bin/sh -c 'echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/pkgs-osrf-archive-keyring.gpg] http://packages.osrfoundation.org/gazebo/ubuntu-stable $(lsb_release -cs) main" | tee /etc/apt/sources.list.d/gazebo-stable.list > /dev/null' \
+  && apt update \
+  && apt install -y ignition-fortress \
+     python3-colcon-common-extensions \
+     python3-colcon-core \
+     python3-rosdep \
+     python3-vcstool \
+  && rm -rf /var/lib/apt/lists/* \
+  && apt clean
+
+# Add the same user as outside the container
+# Requires a docker build argument `USERNAME`
+ARG USERNAME
+
+# Setup working environment
+ENV IGNITION_VERSION fortress
+
+# When running a container start in the user's home folder
+WORKDIR /home/${USERNAME}/
+
+### TODO(quarkytale): creating and building not required for development
+# Create project directory and import packages
+RUN mkdir -p /home/${USERNAME}/buoy_ws/src \
+  && cd /home/${USERNAME}/buoy_ws/src/ \
+  && wget https://raw.githubusercontent.com/osrf/buoy_entrypoint/main/buoy_all.yaml \
+  && vcs import --skip-existing < buoy_all.yaml
+
+WORKDIR /home/${USERNAME}/buoy_ws
+COPY . src/buoy_sim
+
+# Install rosdep dependencies
+RUN apt update \
+  && rosdep update \
+  && rosdep install --from-paths src --ignore-src -r -y -i \
+  && rm -rf /var/lib/apt/lists/* \
+  && apt clean
+
+# Build the project
+RUN /bin/bash -c 'source /opt/ros/${ROS_DISTRO}/setup.bash \
+  && colcon build'
+
+# Source workspaces, add it to the bashrc
+RUN /bin/sh -c 'echo ". /opt/ros/${ROS_DISTRO}/setup.bash" >> ~/.bashrc' \
+  && /bin/sh -c 'echo ". /home/${USERNAME}/buoy_ws/install/setup.sh" >> ~/.bashrc'
```

Comment:
I don't think these lines are having the intended result. After

```
./run.bash -s buoy:latest
```

I'm inside the container and I run

```
$ ign gazebo mbari_wec.sdf
Unable to find or download file
```

So the workspace is not sourced.

Then after sourcing the workspace, I still have some permission issues

```
. ~/buoy_ws/install/setup.bash
ign gazebo mbari_wec.sdf
[Err] [Filesystem.cc:485] Failed to create directory [/home/chapulina/.ignition]: Permission denied
Error opening log file: /home/chapulina/.ignition/gazebo/log/2022-04-27T16:22:30.962065847/server_console.log
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-chapulina'
###ElectroHydraulicPTO subscribing to Double messages on [/model/Hydraulics_Test/joint/HydraulicRam/UserCommandedCurr]
###ElectroHydraulicPTO subscribing to Double messages on [/model/Hydraulics_Test/joint/HydraulicRam/BiasCurrent]
###ElectroHydraulicPTO subscribing to Double messages on [/model/Hydraulics_Test/joint/HydraulicRam/ScaleFactor]
###ElectroHydraulicPTO subscribing to Double messages on [/model/Hydraulics_Test/joint/HydraulicRam/RetractFactor]
[Err] [Filesystem.cc:485] Failed to create directory [/home/chapulina/.ignition]: Permission denied
[Err] [ServerConfig.cc:933] Failed to create directory [/home/chapulina/.ignition/gazebo/6].
```

It looks like `buoy_ws` is created by the root user

```
drwxr-xr-x 1 root root 4096 Apr 27 15:14 buoy_ws
```

Reply:
This is partially solved that the workspace permission is taken care of. But we'd still need to `source /tmp/buoy_ws/install/setup.bash` for the simulation to work. My current lead is the last `RUN` command not having root permission to write in `bashrc` but even after running commands as root user, it's not working as expected. Any thoughts/suggestions?